src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/CSharpPullMemberUpCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp.Dialog;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.PullMemberUp
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(PredefinedCodeRefactoringProviderNames.PullMember)), Shared]
+    internal class CSharpPullMemberUpCodeRefactoringProvider : AbstractPullMemberUpRefactoringProvider
+    {
+        /// <summary>
+        /// For test purpose only.
+        /// </summary>
+        public CSharpPullMemberUpCodeRefactoringProvider(IPullMemberUpOptionsService pullMemberUpService) : base(pullMemberUpService)
+        {
+        }
+
+        internal CSharpPullMemberUpCodeRefactoringProvider() : this(null)
+        {
+        }
+
+        internal override bool IsUserSelectIdentifer(SyntaxNode userSelectedSyntax, CodeRefactoringContext context)
+        {
+            var identifier = GetIdentifier(userSelectedSyntax);
+            return identifier.Span.Contains(context.Span);
```

Comment:
containment isn't probably right here.  for example, if i have   `fo[|ob|]ar`, we probably don't want to offer hte feature.  In *general* we either support:

1. no selection, and the position is on hte name.
2. selection, but the selection has to wrap a node/token fully.  in this case, i would say that if you have a context.span, that it should be larger than the 'span' of the identifier, but smaller than the 'full span' of the identifier.

Reply:
The function you call can return 'default', but i don't see you handling that here.

Reply:
True, I will add checks.